### PR TITLE
Shellrepeater client doesn't handle the specifying the network as a URL Scheme

### PR DESCRIPTION
**Pull request by dchw**
This should surface any future errors properly as well.

The TCP is there to help ensure some consistency in how we specify addresses that connect to external services. This also allows us to implement shared validation for these URLs as well.

cc @alexcb

### CODE REVIEW

File: cmd/earthly/main.go [hunk 1]
```diff
@@ -2629,7 +2629,20 @@ func (app *earthlyApp) actionBuildImp(c *cli.Context, flagArgs, nonFlagArgs []st
...
 	defer cleanCollection.Close()
 
-	go terminal.ConnectTerm(c.Context, app.debuggerHost)
+	go func() {
+		// Dialing doesnt accept URLs, it accepts an address and a "network". These cannot be handled as URL schemes.
+		// Since Shellrepeater hard-codes TCP, we drop it here and log the error if we fail to connect.
+
+		u, err := url.Parse(app.debuggerHost)
+		if err != nil {
+			panic("debugger host was not a URL")
```
Comment (dchw) [inline cmd/earthly/main.go:2638]: This `panic` is consistent with what we do if parsing fails when starting buildkit.

Comment (alexcb) [reply cmd/earthly/main.go:2638]: should we also error out if `u.Scheme != "tcp"`?

File: cmd/earthly/main.go [hunk 2]
```diff
@@ -2629,7 +2629,20 @@ func (app *earthlyApp) actionBuildImp(c *cli.Context, flagArgs, nonFlagArgs []st
...
+		}
+
+		err = terminal.ConnectTerm(c.Context, u.Host)
+		if err != nil {
+			app.console.Warnf("Failed to connect to terminal: %s", err.Error())
```
Comment (alexcb) [inline cmd/earthly/main.go:2643]: :tada: :+1: :1st_place_medal: :ship:

**Review by alexcb — APPROVED** [review]

File: cmd/earthly/main.go (hunk 1 above)
Comment (dchw) [reply cmd/earthly/main.go:2638]: We already do. See the URL validation we do in `before`: https://github.com/earthly/earthly/blob/b3f056b510d822731710f3e463abbc27ed19fcd7/cmd/earthly/main.go#L1111-L1126

Comment (alexcb) [reply cmd/earthly/main.go:2638]: awesome! thanks for pointing that out.

**Pull request merged**